mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -530,7 +530,7 @@ struct LockedAmountInfo {
 }
 
 struct SwapsContext {
-    running_swaps: Mutex<Vec<Weak<dyn AtomicSwap>>>,
+    running_swaps: Mutex<Vec<(Weak<dyn AtomicSwap>, AbortOnDropHandle)>>,
```

Comment:
note that we never delete these weak refs from this list (nor the newly introduced aborthandle).
consider a refactor with different structure for this list, and avoid using weak refs if possible.

Reply:
[245ea93](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2280/commits/245ea93d3174e0187b38d0e56993b810ad82de0f)